From 4023cba52682cd97279ac4ac657111512403780f Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 12 Mar 2018 23:39:10 +0100
Subject: [PATCH 1/1] MAINTAINERS: ARM TI: remove invalid paths

Remove non-existing paths.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 MAINTAINERS | 2 --
 1 file changed, 2 deletions(-)

diff --git a/MAINTAINERS b/MAINTAINERS
index 704ee53e0de..50e82d4d245 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -224,8 +224,6 @@ S:	Maintained
 T:	git git://git.denx.de/u-boot-ti.git
 F:	arch/arm/mach-davinci/
 F:	arch/arm/mach-keystone/
-F:	arch/arm/cpu/arm926ejs/omap/
-F:	arch/arm/cpu/armv7/omap*/
 F:	arch/arm/include/asm/arch-omap*/
 F:	arch/arm/include/asm/ti-common/
 
-- 
2.16.1

