From 0630e878acca31b4dc516218095847c24159c784 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 24 Dec 2017 18:55:59 +0100
Subject: [PATCH 1/1] efi_selftest: execute only 'block device' test

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_selftest/efi_selftest.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/efi_selftest/efi_selftest.c b/lib/efi_selftest/efi_selftest.c
index fc5ef254a1..243f2d6b0a 100644
--- a/lib/efi_selftest/efi_selftest.c
+++ b/lib/efi_selftest/efi_selftest.c
@@ -249,6 +249,7 @@ efi_status_t EFIAPI efi_selftest(efi_handle_t image_handle,
 	if (loaded_image->load_options)
 		testname = (u16 *)loaded_image->load_options;
 
+	testname = L"block device";
 	if (testname) {
 		if (!efi_st_strcmp_16_8(testname, "list") ||
 		    !find_test(testname)) {
-- 
2.15.1

